### DISCUSSION TITLE
Rule for incorrect variable scoping?

### DISCUSSION THREAD

**Discussion by rdatta11** [eslint/eslint#16890] [Q&A] [answered]
I found a bug in my code that effectively looks like:

```
try {
  const demo = "foo";
} catch(err) {
  console.log(demo)
}
```

When my code reaches the catch block it throws an error because I'm using the `demo` constant outside of the scope it was declared. I thought `block-scoped-var` would catch this for me, but it seems like it's only applicable to `var`'s not `const` and `let`'s. Is there an ESlint rule I can use to catch this type of error?

**Comment by mdjermanovic** [comment, marked as answer]
The [no-undef](https://eslint.org/docs/latest/rules/no-undef) rule can catch this error, here's a [Playground Demo](https://eslint.org/play/#eyJ0ZXh0IjoidHJ5IHtcbiAgY29uc3QgZGVtbyA9IFwiZm9vXCI7XG59IGNhdGNoKGVycikge1xuICBjb25zb2xlLmxvZyhkZW1vKVxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tdW5kZWYiOlsiZXJyb3IiXX0sImVudiI6eyJlczYiOnRydWUsImJyb3dzZXIiOnRydWV9fX0=).

**Reply by rdatta11** [reply]
Thank you @mdjermanovic. I had that rule in my `.eslintrc` and it wasn't catching it. This was due to this [issue](https://github.com/eslint/eslint/issues/9330). I thought the rule didn't work as I expected.
